app/commands.py

```diff
@@ -2409,13 +2402,40 @@ async def clan_info(ctx: Context) -> str | None:
 @clan_commands.add(Privileges.UNRESTRICTED)
 async def clan_leave(ctx: Context) -> str | None:
     """Leaves the clan you're in."""
-    if not ctx.player.clan:
+    if not ctx.player.clan_id:
         return "You're not in a clan."
     elif ctx.player.clan_priv == ClanPrivileges.Owner:
         return "You must transfer your clan's ownership before leaving it. Alternatively, you can use !clan disband."
 
-    await ctx.player.clan.remove_member(ctx.player)
-    return f"You have successfully left {ctx.player.clan!r}."
+    clan = await clans_repo.fetch_one(id=ctx.player.clan_id)
+    if not clan:
+        return "You're not in a clan."
+
+    clan_members = await players_repo.fetch_many(clan_id=clan["id"])
+
+    await players_repo.update(ctx.player.id, clan_id=0, clan_priv=0)
+    ctx.player.clan_id = None
+    ctx.player.clan_priv = None
+
+    clan_display_name = f"[{clan['tag']}] {clan['name']}"
+
+    if not clan_members:
+        # no members left, disband clan
+        await clans_repo.delete(clan["id"])
+
+        # announce clan disbanding
+        announce_chan = app.state.sessions.channels.get_by_name("#announce")
+        if announce_chan:
+            msg = f"\x01ACTION disbanded {clan_display_name}."
+            announce_chan.send(msg, sender=ctx.player, to_self=True)
```

Comment:
In master, the disbanding announcement only happens for `!mp disband`, and not the `!mp leave` case where no members remain & the clan is implicitly disbanded. This resolves that gap.